### DISCUSSION TITLE
Migration schema diffing fails with deleteRule: 'set null ("...")'

### DISCUSSION THREAD

**Discussion by Kalos-S** [mikro-orm/mikro-orm#5568] [General]
### Describe the bug

When generating migrations using the provided schema diffing breaks and the foreign key constraint on "parent_id" and "tenant_id" is recreated on every subsequent migration.

### Reproduction

```ts
import {
  Entity,
  OneToOne,
  PrimaryKey,
  type Ref
} from "@mikro-orm/postgresql";

@Entity()
export class Parent {
  @PrimaryKey()
  id!: string;

  @PrimaryKey()
  tenant!: string;
}

@Entity()
export class Child {
  @PrimaryKey()
  id!: string;

  @PrimaryKey()
  tenant!: string;

  @OneToOne(() => Parent, { deleteRule: 'set null ("parent_id")', nullable: true })
  parent?: Ref<Parent> | null;
}
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.5

### Node.js version

20.5.1

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Why do you put `set null ("parent_id")` into that? It should be only `set null`, or not?

**Reply by B4nan** [reply]
Oh right, it's for this thing I guess?

https://www.postgresql.org/message-id/CAF%2B2_SGRXQOtumethpuXhsyU%2B4AYzfKA5fhHCjCjH%2BjQ04WWjA%40mail.gmail.com

**Reply by Kalos-S** [reply]
That's a new feature of Postgres 16 (At the bottom of section 5.4.5 of their docs https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-FK) which allows you to selectively apply the set null to the referencing columns

**Reply by Kalos-S** [reply]
@B4nan any plans for fixing this?

**Reply by B4nan** [reply]
Here you go, ended up with a native support so diffing should work for this too.

5353e6a97cdea224ce19fe3b5745951411a12282

**Reply by Kalos-S** [reply]
Amazing! Thank you

I'm sure you hear this enough but your work on MikroORM is amazing! Probably saved me from a lot of headaches and a ton of time.
